proto/tendermint/abci/types.proto

```diff
@@ -117,27 +121,59 @@ message RequestApplySnapshotChunk {
   string sender = 3;
 }
 
+message RequestPrepareProposal {
+  bytes                   hash                 = 1;
+  tendermint.types.Header header               = 2 [(gogoproto.nullable) = false];
+  // txs is an array of transactions that will be included in a block,
+  // sent to the app for possible modifications.
+  repeated bytes          txs                  = 3;
+  LastCommitInfo          last_commit_info     = 4 [(gogoproto.nullable) = false];
+  repeated Evidence       byzantine_validators = 5 [(gogoproto.nullable) = false];
+  // the modified block cannot exceed this size.
+  int64                   max_bytes            = 6;
+}
+
+message RequestProcessProposal {
+  bytes                   hash                 = 1;
+  tendermint.types.Header header               = 2 [(gogoproto.nullable) = false];
+  repeated bytes          txs                  = 3;
+  LastCommitInfo          last_commit_info     = 4 [(gogoproto.nullable) = false];
+  repeated Evidence       byzantine_validators = 5 [(gogoproto.nullable) = false];
+}
+
+message RequestFinalizeBlock {
+  bytes                   hash                 = 1;
+  tendermint.types.Header header               = 2 [(gogoproto.nullable) = false];
+  repeated bytes          txs                  = 3;
+  LastCommitInfo          last_commit_info     = 4 [(gogoproto.nullable) = false];
+  repeated Evidence       byzantine_validators = 5 [(gogoproto.nullable) = false];
```

Comment:
The `Evidence` struct here isn't actually the `Evidence` struct in `types` but just the name of a validator, their voting power and what they did wrong: either `DuplicateVote` or `LightClientAttack` atm

Reply:
As an aside it really is the same type: `types.Evidence` in TM is not a struct, it's an interface, whose implementations wrap the aforementioned duplicate-vote and light-client evidence structures. It gets unpacked from a message shaped like this, though we don't use the generated type directly.

Reply:
I have to say that something about just calling this field `evidence` is appealing to me. Probably due to the fact that we're listing specific evidence of malfeasance (e.g. the height at which the malfeasance took place), and we're not just listing byzantine validators' information.

If we were just capturing byzantine validators' info, then the field name would make more sense to me. But then, of course, it wouldn't be evidence :)

Also applies to the similarly named field in `RequestProcessProposal`.

Reply:
@thanethomson if you have a strong opinion on this we can change it. The main reason it's not clear to me that we should is that the same parameter with the same name has been around in ABCI for years (in `RequestBeginBlock`)... but since these changes are breaking compatibility anyway I guess renaming it is also a good option.

Let me know what option you prefer

Reply:
What I was meaning in my above comment was that the term `Evidence` for me should contain some data that proves that a validator did something wrong (like `types.DuplicateVoteEvidence` does). `abci.Evidence` doesn't show any proof, rather it's telling that app this is what happened i.e. it's a record of misbehavior (in fact if you look at the SDK they call this `Misbehavior`. So perhaps we could rename the struct to `abci.Misbehavior` and call the field `misbehavior`

Reply:
OK, if the SDK is calling it misbehavior, then I think it's a good idea to align with that. I'll rename it today. Please voice any concerns you may have with renaming to "misbehavior"

Reply:
Actually, I started the renaming but I see that (preexisting) `RequestBeginBlock` is also affected, so I would like to defer any action on this until we have completed the `FinalizeBlock` logic in the code